src/Elastic.Apm/Model/Span.cs

```diff
@@ -265,5 +275,40 @@ public void CaptureError(string message, string culprit, StackFrame[] frames, st
 				_enclosingTransaction,
 				parentId ?? (ShouldBeSentToApmServer ? null : _enclosingTransaction.Id)
 			);
+
+		private void DeduceDestination()
+		{
+			if (!_context.IsValueCreated) return;
+
+			if (Context.Http != null) CopyMissingProperties(DeduceHttpDestination());
+
+			void CopyMissingProperties(Destination src)
+			{
+				if (src == null) return;
+
+				if (Context.Destination == null)
+					Context.Destination = src;
+				else
+					Context.Destination.CopyMissingPropertiesFrom(src);
+			}
+		}
+
+		internal Destination DeduceHttpDestination()
+		{
+			try
+			{
+				var url = Context.Http.OriginalUrl ?? new Uri(Context.Http.Url);
```

Comment:
I had a rant about this above, I'm not sure I understand the intention on how the `OriginalUrl` should work, but I guess this allocation could be avoided, we already do this for the sanitization.

Reply:
I've added some of the code in `Sanitize` (the code to which I referred as "redundant") back to keep to avoid `Uri` allocation. Are you sure this optimization worth the complication?

Reply:
This is fine. I think with your change `Context.Http.OriginalUrl` is always not `null`, but yeah, I'd keep this line - this look ok to me.